native/src/main/java/io/ballerina/lib/milvus/Utils.java

```diff
@@ -79,4 +91,91 @@ private static Object convertToJsonFields(Object value) {
         }
         return value;
     }
+
+    protected static void generateSearchResult(RecordType outputFieldsType, List<SearchResp.SearchResult> results,
+                                             BMap<BString, Object>[] responses) {
+        for (SearchResp.SearchResult result : results) {
+            BMap<BString, Object> response =
+                    ValueCreator.createRecordValue(ModuleUtils.getModule(), SEARCH_RESULT);
+            BMap<BString, Object> entity = ValueCreator.createRecordValue(outputFieldsType);
+            getResult(result, entity);
+            response.put(PRIMARY_KEY, StringUtils.fromString(result.getPrimaryKey()));
+            response.put(SEARCH_ID, result.getId());
+            response.put(SIMILARITY_SCORE, result.getScore().doubleValue());
+            response.put(OUTPUT_FIELDS, entity);
+            responses[results.indexOf(result)] = response;
+        }
+    }
+
+    protected static void generateQueryResult(List<QueryResp.QueryResult> results,
+                                              BMap<BString, Object>[] responses) {
+        for (QueryResp.QueryResult result : results) {
+            BMap<BString, Object> response =
+                    ValueCreator.createRecordValue(ModuleUtils.getModule(), QUERY_RESULT);
+            getResult(result, response);
+            responses[results.indexOf(result)] = response;
+        }
+    }
+
+    private static void getResult(SearchResp.SearchResult result, BMap<BString, Object> entity) {
+        for (String key: result.getEntity().keySet()) {
+            if (result.getEntity().get(key) instanceof List<?> list) {
+                if (!list.isEmpty() && list.get(0) instanceof Number) {
+                    double[] values = list.stream()
+                            .mapToDouble(value -> ((Number) value).floatValue())
+                            .toArray();
+                    BArray floatArray = ValueCreator.createArrayValue(values);
+                    entity.put(StringUtils.fromString(key), floatArray);
+                } else {
+                    BString[] values = list.stream()
+                            .map(value -> StringUtils.fromString(value.toString()))
+                            .toArray(BString[]::new);
+                    BArray stringArray = ValueCreator.createArrayValue(values);
+                    entity.put(StringUtils.fromString(key), stringArray);
+                }
+            } else {
+                entity.put(StringUtils.fromString(key),
+                        StringUtils.fromString(result.getEntity().get(key).toString()));
+            }
+        }
+    }
+
+    private static void getResult(QueryResp.QueryResult result, BMap<BString, Object> entity) {
```

Comment:
It seems both this method and the earlier one are overridden methods with similar logic. Should we create a new overridden version of getResult that accepts an `Map.Entry` as the first parameter, and then invoke this new method from the existing ones?